bin/check_hcid.py

```diff
@@ -0,0 +1,249 @@
+            if counts[ref] > 0:
+                hcid_dict[taxon]["mapped_required"] += 1
+                hcid_dict[taxon]["mapped_count"] += counts[ref]
+            hcid_dict[taxon]["mapped_required_details"].append("%s:%s" %(ref,str(counts[ref])))
+        hcid_dict[taxon]["mapped_required"] = float(hcid_dict[taxon]["mapped_required"])/len(hcid_dict[taxon]["required_refs"])
+        hcid_dict[taxon]["mapped_required_details"] = "|".join(hcid_dict[taxon]["mapped_required_details"])
+        for ref in hcid_dict[taxon]["additional_refs"]:
+            if counts[ref] > 0:
+                hcid_dict[taxon]["mapped_additional"] += 1
+                hcid_dict[taxon]["mapped_count"] += counts[ref]
+        if len(hcid_dict[taxon]["additional_refs"])>0:
+            hcid_dict[taxon]["mapped_additional"] = float(hcid_dict[taxon]["mapped_additional"])/len(hcid_dict[taxon]["additional_refs"])
+
+
+def report_findings(hcid_dict, prefix):
+    found = []
+    for taxid in hcid_dict:
+        if hcid_dict[taxid]["mapped_found"] and (hcid_dict[taxid]["classified_found"] or hcid_dict[taxid]["classified_parent_found"]) :
+            with open("%s.warning" %taxid, "w") as f_warn:
+                f_warn.write(f"WARNING: Found {hcid_dict[taxid]['classified_count']} classified reads ({hcid_dict[taxid]['mapped_count']} mapped reads) of {hcid_dict[taxid]['name']}\n")
+            found.append(taxid)
+    keys = ["name", "taxon_id", "min_count", "classified_count","mapped_count", "mapped_required", "mapped_required_details", "mapped_additional"]
+    with open("%s.counts.csv" %prefix,'w') as f_counts:
+        w = csv.DictWriter(f_counts,keys)
+        w.writeheader()
+        for taxid in hcid_dict:
+            w.writerow({key: hcid_dict[taxid][key] for key in keys})
+    return found
+
+# Main method
+def main():
+    # Parse arguments
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        "-k",
+        dest="kreport_file",
+        required=False,
+        help="Kraken or Bracken file of taxon relationships and quantities",
+    )
+    parser.add_argument(
+        "-r",
+        dest="reads",
+        required=True,
+        help="FASTQ of reads",
+    )
+    parser.add_argument(
+        "-t",
+        dest="taxonomy",
+        required=False,
+        help="Taxonomy directory containing the nodes.dmp file. If not provided will infer from report file but this may lead to fewer reads extracted",
+    )
+    parser.add_argument(
+        "-i",
+        dest="hcid_file",
+        required=False,
+        default="resources/hcid.json",
+        help="JSON file specifying HCID to look for a min counts to notify",
+    )
+    parser.add_argument(
+        "-p",
+        dest="prefix",
+        required=False,
+        default="hcid",
+        help="Output prefix",
+    )
+    parser.add_argument(
+        "-d",
+        dest="ref_fasta",
+        required=False,
+        default="resources/hcid_refs.fq.gz",
```

Comment:
Probably aught to be `resources/hcid_refs.fq.gz`

Reply:
The default here suggests it should be a `fq.gz` :)

Reply:
fixed